`PurchasesOrchestrator`: fixed callback not invoked regression during downgrades

NachoSoto opened the pull request:
Fixes #3020.
This essentially reverts #2890.

As shown in #3020, downgrades are reported with the same transaction identifier as the original transaction. Therefore our approach in #2890 is invalid.
We considered checking whether the `SKPayment` in the `SKPaymentTransaction` matches the one added to the queue as a way to disambiguate these transactions, but it doesn't. So we have no way of telling the difference between queue transactions and purchases.

I've left the original tests as a way to at least document this behavior.

Abbott-Deng commented:
@NachoSoto I upgrade to 4.25.5, I get one error with "Value of type 'Purchases' has no member 'storeKit2Setting'",  I found the storeKit2Setting is only visible in DEBUG macro. Please help check it.

NachoSoto commented:
Thanks @Abbott-Deng. Sorry for the troubles.

I just fixed that in #3034 and will make a new release. I also added a new CI job that builds the SDK in release mode using `Xcode 15 beta` so we can catch this in the future.

NachoSoto commented:
That's fixed in 4.25.6. Thanks again!
